fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReverseChainTest.java

```diff
@@ -0,0 +1,1034 @@
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter 'badSearchParm' for resource type 'Procedure' was not found.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmNotFoundError2() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:badSearchParm", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter 'badSearchParm' for resource type 'Procedure' was not found.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmNotReferenceError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:code:code", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter 'code' is not of type reference for '_has' (reverse chain) search.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmTargetTypeNotValidError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:encounter:code", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter 'encounter' target types do not include expected type 'Patient' for '_has' (reverse chain) search.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmUndefinedModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:code:badModifier", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Undefined Modifier: badModifier");
```

Comment:
Refer to corresponding comment in SearchUtil (this response should have 'badModifier' and possibly be URL Encoded.

Reply:
done